src/commands/roleCommands/changecolor.ts

```diff
@@ -283,25 +284,22 @@ export class ColorRoles {
           position: rolePosition,
           mentionable: false,
         }))
+      await member.roles.add(colorRole).catch(console.error)
+    }
 
-      // Remove and delete existing role if exists
-      const existingRole = member.roles.cache.find((role) => ColorRoles.hexExp.test(role.name))
-      return member.roles.add(colorRole).then(async (guildMember) => {
-        if (existingRole) {
-          member.roles.remove(existingRole).catch(console.error)
-          const roleToDelete = guild.roles.cache.find((role) => role.id === existingRole.id)
-          if (
-            roleToDelete &&
-            (roleToDelete.members.size === 0 ||
-              (roleToDelete.members.size === 1 && roleToDelete.members.some((_, id) => id === member.id)))
-          ) {
-            guild.roles.delete(existingRole.id).catch(console.error)
-          }
-        }
-
-        return guildMember
-      })
+    if (existingRole) {
+      member.roles.remove(existingRole).catch(console.error)
+      const roleToDelete = guild.roles.cache.find((role) => role.id === existingRole.id)
+      if (
+        roleToDelete &&
+        (roleToDelete.members.size === 0 ||
+          (roleToDelete.members.size === 1 && roleToDelete.members.some((_, id) => id === member.id)))
```

Comment:
Now that new role gets added first there's no need to check if the member is about to receive it but we should await the removal of the role so the number of members using the role is accurate.

Reply:
Yeah this was pretty messy and before I understood promises a bit. I've cleaned this up and it should be more clear at what is actually happening.